### PR TITLE
Feat/settings screen

### LINKED ISSUE #14: Make Settings Page
### Description
A settings screen needs to be created that allows users to update their profile information. This includes changing their name, uploading a logo or profile picture, and editing their description.

### Requirements
- **Input Fields:**
  - **Name:** A text input field for the user to enter or update their name.
  - **Logo/Profile Picture:** An upload input that allows users to select and upload an image file for their logo or profile picture.
  - **Description:** A text area for users to provide or update a brief description about themselves or their business.
- **Responsive Design:** Ensure that the layout is responsive and user-friendly across different devices.
- **Validation:** Implement basic validation for required fields (e.g., name).

### Technical Considerations
- Use **TailwindCSS** for styling and layout to maintain consistency with the rest of the application.
- Ensure that components are implemented using **shadcn/ui** and **Radix UI** for enhanced user experience.

### Tasks
1. Design the layout of the settings screen with appropriate input fields.
2. Implement each input field with necessary props and validation logic.

### Notes
This settings screen will allow users to personalize their profiles and improve their experience within the application. This is just for design, save buttons should not store any input for now.

### PR DESCRIPTION
**Pull request by Villarley**
# Pull Request Overview

## 📝 Summary
Users can:

Edit their name.
Upload or change their profile picture (with preview functionality).
Update their personal or business description.
View a responsive design suitable for different devices.
### 🔗 Related Issues
- Closes #14 

## 🔄 Changes Made
Frontend Implementation:

Designed a new SettingsScreen component using React, TailwindCSS, and Next.js.
Added components for:

1. Name input (Input).
2. Profile picture upload with real-time preview.
3. Description editor (Textarea).
4. Dynamic edit/save functionality for user convenience.
Responsive Design:

Optimized layout for mobile, tablet, and desktop screens.
Ensured seamless interaction across devices.
Error Handling:

Basic validation for required fields like Name.
Error messages displayed inline when validation fails.

## 🖼️ Current Output
image
image

## 💬 Comments
Thank you for the opportunity of contributing, looking forward to keep contributing!!

### CODE REVIEW

File: paystell-frontend/src/app/settings/page.tsx [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+                width={120}
+                height={120}
+                className="rounded-full border"
+              />
+              <button
+                type="button"
+                onClick={() =>
+                  document.getElementById("logoInput")?.click()
+                }
+                className="absolute bottom-0 right-0 bg-blue-500 text-white p-2 rounded-full"
```
Comment (MPSxDev) [inline paystell-frontend/src/app/settings/page.tsx:62]: use bg-primary
```suggestion
                className="absolute bottom-0 right-0 bg-primary text-white p-2 rounded-full"
```

File: paystell-frontend/src/app/settings/page.tsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+                </Button>
+              </>
+            ) : (
+              <Button
+                type="button"
+                className="bg-blue-500 w-full"
+                onClick={() => setIsEditing(true)}
+              >
+                Edit Profile
+              </Button>
```
Comment (MPSxDev) [inline paystell-frontend/src/app/settings/page.tsx:128]: remove className, use variant 
```suggestion
              <Button
                type="button"
                variant="default"
                onClick={() => setIsEditing(true)}
              >
                Edit Profile
              </Button>
```

File: paystell-frontend/src/app/settings/page.tsx [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+                  onClick={() => setIsEditing(false)}
+                >
+                  Cancel
+                </Button>
+                <Button
+                  type="submit"
+                  className="bg-blue-500"
+                >
+                  Save
+                </Button>
```
Comment (MPSxDev) [inline paystell-frontend/src/app/settings/page.tsx:119]: use variant
```suggestion
                <Button
                  type="submit"
                  variant="default"
                >
                  Save
                </Button>
```

File: paystell-frontend/src/app/settings/page.tsx [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+          <div className="flex justify-between">
+            {isEditing ? (
+              <>
+                <Button
+                  type="button"
+                  className="bg-gray-500"
+                  onClick={() => setIsEditing(false)}
+                >
+                  Cancel
+                </Button>
```
Comment (MPSxDev) [inline paystell-frontend/src/app/settings/page.tsx:113]: use variant 
```suggestion
                <Button
                  type="button"
                  variant="destructive"
                  onClick={() => setIsEditing(false)}
                >
                  Cancel
                </Button>
```

File: paystell-frontend/package-lock.json
Comment (MPSxDev) [inline paystell-frontend/package-lock.json:1]: remove file package-lock.json

File: paystell-frontend/package.json [hunk 5]
```diff
@@ -9,13 +9,16 @@
...
+    "@radix-ui/react-label": "^2.1.0",
+    "@radix-ui/react-slot": "^1.1.0",
     "@radix-ui/themes": "^3.1.5",
     "class-variance-authority": "^0.7.0",
     "clsx": "^2.1.1",
     "lucide-react": "^0.456.0",
     "next": "14.2.17",
     "react": "^18",
     "react-dom": "^18",
+    "shadcn-ui": "^0.9.3",
```
Comment (MPSxDev) [inline paystell-frontend/package.json:21]: Update 2 high severity vulnerabilities
```suggestion
    "shadcn-ui": "^0.2.3",
```

**Review by MPSxDev — CHANGES_REQUESTED** [review]
good job just some minor corrections

File: paystell-frontend/src/app/settings/page.tsx
Comment (MPSxDev) [inline paystell-frontend/src/app/settings/page.tsx:1]: add the content to the new file deployed in dashboard/settings

**Review by MPSxDev — CHANGES_REQUESTED** [review]

**Comment by Villarley** [comment]
@MPSxDev ready sir!

**Review by MPSxDev — APPROVED** [review]
Great job. Thanks @Villarley

**Pull request merged**